examples/convert_ieeg_to_bids.py

```diff
@@ -70,65 +67,63 @@
+# transform to get to this space from "mri" or Freesurfer surface RAS space.
+# So, first we have to use the ``trans`` we found from assigning or estimating
+# fiducial points (see `Locating Intracranial Electrode Contacts
+# <https://mne.tools/dev/auto_tutorials/clinical/10_ieeg_localize.html>`_),
+# and then we can use the Freesurfer Talairach transform to get to MNI space.
 
-# %%
-# We will load a :class:`mne.io.Raw` object and
-# use the montage we created.
-info = mne.create_info(ch_names, 1000., 'ecog')
-raw = mne.io.read_raw_edf(misc_path + '/ecog/sample_ecog.edf')
-raw.info['line_freq'] = 60  # specify power line frequency as required by BIDS
-raw.set_channel_types({ch: 'ecog' for ch in raw.ch_names})
+# estimate trans
+lpa, nasion, rpa = mne.coreg.get_mni_fiducials('sample_seeg', subjects_dir)
+montage = mne.channels.make_dig_montage(
+    lpa=lpa['r'], nasion=nasion['r'], rpa=rpa['r'],
+    coord_frame='mri')
+trans = mne.channels.compute_native_head_t(montage)
 
-# set the bad channels
-raw.info['bads'].extend(['BTM1', 'BTM2', 'BTM3', 'BTM4', 'BTM5', 'BTM6',
-                         'BTP1', 'BTP2', 'BTP3', 'BTP4', 'BTP5', 'BTP6',
-                         'EKGL', 'EKGR'])
+# get Talairach transform
+mri_mni_t = mne.read_talxfm('sample_seeg', subjects_dir)
 
-# set montage
-# (note that this only works for some channel types: EEG/sEEG/ECoG/DBS/fNIRS)
-raw.set_montage(montage, on_missing='warn')
 
 # %%
 # Let us confirm what our channel coordinates look like.
 
-# make a plot of the sensors in 2D plane
-raw.plot_sensors(ch_type='ecog')
+fig_kwargs = dict(size=(800, 600), bgcolor='w', scene=False)
+renderer = mne.viz.backends.renderer.create_3d_figure(**fig_kwargs)
+al_kwargs = dict(
+    show_axes=True, surfaces=dict(pial=0.2), coord_frame='mri',
+    subjects_dir=subjects_dir)
+fig = mne.viz.plot_alignment(raw.info, trans, 'sample_seeg',
+                             fig=renderer.figure, **al_kwargs)
+view_kwargs = dict(azimuth=60, elevation=100, distance=0.3)
+mne.viz.set_3d_view(fig, **view_kwargs)
 
-# Get the first 5 channels and show their locations.
-picks = mne.pick_types(raw.info, ecog=True)
-dig = [raw.info['dig'][pick] for pick in picks]
-chs = [raw.info['chs'][pick] for pick in picks]
-pos = np.array([ch['r'] for ch in dig[:5]])
-ch_names = np.array([ch['ch_name'] for ch in chs[:5]])
-print("The channel coordinates before writing into BIDS: ")
-pprint([x for x in zip(ch_names, pos)])
+# %%
+# Now let's convert the montage to MNI Talairach ("mni_tal").
+montage = raw.get_montage()
+montage.apply_trans(mne.transforms.invert_transform(trans))  # head->mri
+montage.apply_trans(mri_mni_t)
+# a bit of a hack here; MNE will transform the coordinates to "head"
+# when you set the montage if there are fiducials and we don't want
+# that so we have to get rid of them
+montage.dig = montage.dig[3:]
+# warns that identity transformation to "head" is assumed which is what we want
+raw.set_montage(montage, verbose='error')
```

Comment:
we need to discuss this on discord with @larsoner

Reply:
this gymnastic is required by write_raw_bids for sEEG data?

Reply:
Yes and no. Yes, the estimated fiducials and head->mri `trans` are more easily found this way for sEEG because there is no reason to have fiducials in your pipeline for sEEG so it makes sense to do them in an automated fashion. No, the transform to "head" is used properly, the only reason it is not in this instance (and wasn't before this PR, this PR just makes it explicit that we're suppressing the transformation to "head") is because we want to only pass a `raw` object with coordinate locations but those locations cannot be in "head" (unless we also supply a `trans` and an `mri_mni_t` (Talairach). I had it that way originally but it is very cumbersome and those arguments are unused for non-ieeg data.

Reply:
Related issue I had a long time ago when things were just mapping from "unknown" to "head" within mne-python:

https://github.com/mne-tools/mne-python/issues/7691

Reply:
Yeah I read the issue and I agree that ideally you wouldn't use "head" coordinates but like I said in another post, it's not the worst-defined coordinate frame because it's scan-agnostic (again in sEEG the lack of accuracy doesn't really make this worthwhile but I think in principle it's not too bad and it's not too bad to explain in tutorials to users). Especially if you can use the estimate from fsaverage fiducials, I think it's an okay solution.